packages/suite/src/actions/wallet/selectedAccountActions.ts

```diff
@@ -158,6 +166,8 @@ const getAccountState = (state: AppState): SelectedAccountStatus => {
                     network,
                     discovery,
                     params,
+                    mode,
+                    account: account || undefined,
```

Comment:
@marekrjpolak I would like to know your opinion regarding this change I think it is required in order to know that it is a `coinjoin` account in `getAccountStateWithMode` when the discovery fails.

Imagine this situation:


   1. Start coinjoin account
   2. Remember wallet
   3.  Disable Tor during discovery
   4.  Refresh the app (ctrl + r)
   5.  Go to The remembered coinjoin account that should give error with discovery since Tor is disable (you should have "devmode" off)

In step 5 I find that `selectedAccount` is in it's initial status {status: 'none'} because it was not return here.

So adding `account` to this return fixes that.

Reply:
My discovery is able to continue and finish after CTRL+R even if I disable Tor during discovery. Is that expected?

Other than that, I think that adding `account` is acceptable, but not necessary, because it's (given that `status` is `'exception'`) then used only in `getAccountStateWithMode` and we know exactly that the account is `coinjoin` because we're on line 169.

So my proposal is
a) to use `selectedAccount` param instead of `state.wallet.selectedAccount` in `getAccountStateWithMode` which should work exactly the same for successfully `loaded` accounts (on line 184), and maybe even without the double-updating from my comment below, and
b) add the `tor-disable` mode separately (if needed) on line 169 without adding the `account`. Otherwise it's not only used in `getAccountStateWithMode` but also stored in the selectedAccount reducer even if status is 'exception' which could theoretically break something.

But take it only as recommendations :) I'd really like to refactor this `getAccountState` and `getAccountStateWithMode` somehow in the future so it's more obvious what it does.

Reply:
> My discovery is able to continue and finish after CTRL+R even if I disable Tor during discovery. Is that expected?

Yes if you are running in dev mode, you should change `packages/suite-desktop/src/modules/request-interceptor.ts`  `isDevEnv` to be `false` to make it fail. I should fail at the `request-manager` level so it really should no continue discovery when Tor is disable.

I followed your advice `add the tor-disable mode separately (if needed) on line 169 without adding the account.` and it works fine. 

In `getAccountStateWithMode` I changed to use the `selectedAccount` argument in the function for cases when the discovery is successful but Tor is disable.